src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs

```diff
@@ -73,22 +77,22 @@ public string GetOriginalMessage(IReadOnlyList<KeyValuePair<string, object>> mes
         }
 
         /// <summary>
-        /// Verify that the input parameterList contains non-empty key-values and the orignal-format-property at the end
+        /// Verify that the input parameterList contains non-empty key-values and the original-format-property at the end
         /// </summary>
-        private static bool IsValidParameterList(IReadOnlyList<KeyValuePair<string, object>> parameterList, out int? originalMessageIndex, out bool hasMessageTemplateCapture, out bool isMixedPositional, out bool isPositional)
+        /// <returns>null if not valid parameter list</returns>
+        private static ParameterListAnalyisResult? IsValidParameterList(IReadOnlyList<KeyValuePair<string, object>> parameterList)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)